src/main/kotlin/com/help/howtosecurity/config/SecurityConfig.kt

```diff
@@ -11,80 +13,68 @@ import org.springframework.security.config.web.server.ServerHttpSecurity
-                    canAccess(authentication, "dummy")
-                }
-            }
             .and().build()
     }
 
-//    @Bean
-//    @Order(1)
-//    fun permitAllWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain? {
-//        return http
-//            .authorizeExchange()
-//            .pathMatchers("/permitall/**").permitAll()
-//            .and().build()
-//    }
-
-//    @Bean
-//    @Order(2)
-//    fun basicAuthWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain? {
-//        return http
-//            .httpBasic().authenticationManager { basicAuthenticationDummyAuthentication(it) }.and()
-//            .authorizeExchange()
-//            .pathMatchers("/securebasic/**").access {
-//                authMono, _ -> authMono.map { authentication ->
-//                    canAccess(authentication, "basic")
-//                }
-//            }
-//            .and().build()
-//    }
-
-//    @Bean
-//    @Order(3)
-//    fun oAuthWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain? {
-//        return http
-//            .oauth2ResourceServer().jwt().and().and()
-//            .authorizeExchange()
-//            .pathMatchers("/secureoauth/**").authenticated()
-//            .and().build()
-//    }
+    @Bean
+    @Order(2)
+    fun basicAuthWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain? {
+        return http
+            .securityMatcher(pathMatchers("/securebasic/**"))
+            .httpBasic().authenticationManager { basicAuthenticationDummyAuthentication(it) }.and()
+            .authorizeExchange { spec ->
+                run {
+                    spec.pathMatchers("/securebasic/**").authenticated()
+                }
+            }
+            .exceptionHandling()
+            .authenticationEntryPoint(HttpStatusServerEntryPoint(HttpStatus.UNAUTHORIZED))
+            .and()
+            .build()
+    }
 
-    private fun canAccess(authentication: Authentication, nameToCompare: String): AuthorizationDecision {
-        when(authentication) {
-            is OAuth2AuthenticationToken -> {
-                if ((authentication.principal as DefaultOidcUser).userInfo.givenName == nameToCompare) {
-                    return AuthorizationDecision(true)
+    @Bean
+    @Order(3)
+    fun oAuthWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
+        return http
+            .httpBasic().disable()
+            .securityMatcher(pathMatchers("/secureoauth/**"))
+            .authorizeExchange { spec ->
+                run {
+                    spec.pathMatchers("/secureoauth/**").authenticated()
                 }
             }
-        }
-        if (authentication.name == nameToCompare) {
-            return AuthorizationDecision(true)
-        }
-        return AuthorizationDecision(false)
+            .oauth2ResourceServer(
+                ServerHttpSecurity.OAuth2ResourceServerSpec::jwt
+            )
+            .exceptionHandling()
```

Comment:
In this use case since we are using the Okta client, if we comment this Exception Handling you will be redirected to Okta to login